tests/e2e/gc_rwx_volume_health.go

```diff
@@ -0,0 +1,339 @@
+		framework.Logf("volume Name:%s, capacity:%d volumeType:%s health:%s accesspoint: %s",
+			queryResult.Volumes[0].Name,
+			queryResult.Volumes[0].BackingObjectDetails.(*cnstypes.CnsVsanFileShareBackingDetails).CapacityInMb,
+			queryResult.Volumes[0].VolumeType, queryResult.Volumes[0].HealthStatus,
+			queryResult.Volumes[0].BackingObjectDetails.(*cnstypes.CnsVsanFileShareBackingDetails).AccessPoints)
+
+		ginkgo.By("Verifying volume type specified in PVC is honored")
+		gomega.Expect(queryResult.Volumes[0].VolumeType == testVolumeType).To(gomega.BeTrue(),
+			"Volume type is not FILE")
+		ginkgo.By("Verifying volume size is honored")
+		newSizeInMb := int64(2048)
+		gomega.Expect(queryResult.Volumes[0].BackingObjectDetails.(*cnstypes.CnsVsanFileShareBackingDetails).
+			CapacityInMb == newSizeInMb).To(gomega.BeTrue(), "Volume Capaticy is not matching")
+
+		if volHealthCheck {
+			ginkgo.By("poll for health status annotation")
+			err = pvcHealthAnnotationWatcher(ctx, client, pvclaim, healthStatusAccessible)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		}
+
+		//Add a check to validate CnsVolumeMetadata crd
+		err = waitAndVerifyCnsVolumeMetadata4GCVol(fcdIDInCNS, pvcNameInSV, pvclaim, pv, nil)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+	})
+
+	/*
+	   Annotations should be accessble to all the RWX PVCs which are in bound state.
+	   1. Create a PVC using invalid SC
+	   2. PVC should be in pending state
+	   3. Verify health annotation is not added on the PVC
+	   4. Delete PVC
+	*/
+	ginkgo.It("Verify volume health annotation is not added for PVC in pending", func() {
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+		var storageclasspvc *storagev1.StorageClass
+		var pvclaim *v1.PersistentVolumeClaim
+		var err error
+		ginkgo.By("CNS_TEST: Running for GC setup")
+		scParameters[svStorageClassName] = nonVsanStoragePolicyName
+
+		ginkgo.By("Creating a PVC")
+		storageclasspvc, pvclaim, err = createPVCAndStorageClass(client,
+			namespace, nil, scParameters, diskSize, nil, "", false, v1.ReadWriteMany)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		defer func() {
+			err = client.StorageV1().StorageClasses().Delete(ctx, storageclasspvc.Name, *metav1.NewDeleteOptions(0))
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		}()
+
+		ginkgo.By("Expect claim to fail as the storage policy mentioned in Storage class " +
+			"has non-vSAN compliant datastores")
+		err = fpv.WaitForPersistentVolumeClaimPhase(v1.ClaimBound, client,
+			pvclaim.Namespace, pvclaim.Name, framework.Poll, time.Minute/2)
+		gomega.Expect(err).To(gomega.HaveOccurred())
+
+		defer func() {
+			err = fpv.DeletePersistentVolumeClaim(client, pvclaim.Name, pvclaim.Namespace)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		}()
+
+		ginkgo.By(fmt.Sprintf("Sleeping for %v minutes to allow volume health check to be triggered",
+			healthStatusWaitTime))
+		time.Sleep(healthStatusWaitTime)
```

Comment:
this is set to 2mins, will the health get updated in that time?
I think we should add a watch here and set min and max wait-times with periodic intervals

Reply:
Health status interval is set to 2 min by Jenkins pre-setup part - so is the automation code.

The function at line 332 pvcHealthAnnotationWatcher watches periodically for health update